pkg/infrastructure/gcp/clusterapi/network.go

```diff
@@ -85,9 +105,72 @@ func createInternalLBAddress(ctx context.Context, in clusterapi.InfraReadyInput)
 		},
 	}
 
-	if _, err := service.HealthChecks.Insert(in.InstallConfig.Config.GCP.ProjectID, healthCheck).Context(ctx).Do(); err != nil {
+	_, err = service.RegionHealthChecks.Insert(projectID, region, healthCheck).Context(ctx).Do()
+	if err != nil {
 		return "", fmt.Errorf("failed to create api-internal health check: %w", err)
 	}
 
+	if err := WaitForOperationRegional(ctx, projectID, region, op); err != nil {
+		return "", fmt.Errorf("failed to wait for health check creation: %w", err)
+	}
+
+	hc, err := service.RegionHealthChecks.Get(projectID, region, hcName).Context(ctx).Do()
+	if err != nil {
+		return "", fmt.Errorf("error getting health check: %w", err)
+	}
+	var backends []*compute.Backend
+	for _, zone := range zones {
+		igName := fmt.Sprintf("%s-apiserver-%s", in.InfraID, *zone)
+		ig, err := service.InstanceGroups.Get(projectID, *zone, igName).Context(ctx).Do()
+		if err != nil {
+			return "", fmt.Errorf("error getting instance group %s in zone %s: %w", igName, *zone, err)
+		}
+		backends = append(backends, &compute.Backend{
+			BalancingMode: "CONNECTION",
+			Group:         ig.SelfLink,
+		})
+	}
+
+	besvcName := fmt.Sprintf("%s-api-internal", in.InfraID)
+	op, err = service.RegionBackendServices.Insert(projectID, region, &compute.BackendService{
+		Backends:            backends,
+		Name:                besvcName,
+		LoadBalancingScheme: "INTERNAL",
+		Protocol:            "TCP",
+		TimeoutSec:          int64((10 * time.Minute).Seconds()),
+		HealthChecks:        []string{hc.SelfLink},
+		Region:              region,
+		Network:             networkSelfLink,
+	}).Context(ctx).Do()
+	if err != nil {
+		return "", fmt.Errorf("failed to create internal backend service: %w", err)
+	}
+
+	if err := WaitForOperationRegional(ctx, projectID, region, op); err != nil {
+		return "", fmt.Errorf("failed to wait for internal backend service creation: %w", err)
+	}
+
+	besvc, err := service.RegionBackendServices.Get(projectID, region, besvcName).Context(ctx).Do()
+	if err != nil {
+		return "", fmt.Errorf("failed to get backend service: %w", err)
+	}
+
+	op, err = service.ForwardingRules.Insert(projectID, region, &compute.ForwardingRule{
+		Name:                fmt.Sprintf("%s-api-internal", in.InfraID),
+		IPProtocol:          "TCP",
```

Comment:
Also need to set
`IPAddress:           ipAddress`
in order to associate this rule with the correct IP.